internal/errors/errors.go

```diff
@@ -0,0 +1,26 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package errors
+
+var (
+	// ErrorCodeUnknown is a generic error that can be used as a last
+	// resort if there is no situation-specific error message that can be used
+	ErrorCodeUnknown = Register("errcode", ErrorDescriptor{
```

Comment:
This is an advanced error system. However, I have concerns that each time we use the `ErrorCode.Error()` and other methods, we will call `ErrorCode.Descriptor` and trigger a map matching, which may provide additional overhead. Why do we need the map from error code to error descriptor? Can we merge the error code and error descriptor to simplify the design? Maybe I'm wrong, please correct me.